packages/perseus/src/widgets/numeric-input/numeric-input.tsx

```diff
@@ -1,419 +1,123 @@
-    ) => void = (path, newValue, cb) => {
-        /* c8 ignore next */
-        this.props.onChange(
-            {
-                currentValue: newValue,
+import {type NumericInputProps} from "./numeric-input.class";
+import {generateExamples, shouldShowExamples} from "./utils";
+
+type InputRefType = SimpleKeypadInput | InputWithExamples | null;
+
+/**
+ * The NumericInputComponent is a child component of the NumericInput class
+ * component. It is responsible for rendering the UI elements of the Numeric
+ * Input widget.
+ */
+export const NumericInputComponent = forwardRef(
+    (props: NumericInputProps, ref) => {
+        const context = useContext(PerseusI18nContext);
+        const inputRef = useRef<InputRefType>(null);
+        const [isFocused, setIsFocused] = useState<boolean>(false);
+
+        // Pass the focus and blur methods to the Numeric Input Class component
+        useImperativeHandle(ref, () => ({
+            focus: () => {
+                if (inputRef.current) {
+                    inputRef.current.focus();
+                    setIsFocused(true);
+                }
             },
-            cb,
-        );
-    };
-
-    getUserInput(): PerseusNumericInputUserInput {
-        return NumericInput.getUserInputFromProps(this.props);
-    }
-
-    getPromptJSON(): NumericInputPromptJSON {
-        return _getPromptJSON(this.props, this.getUserInput());
-    }
-
-    handleChange: (
-        arg1: string,
-        arg2?: () => unknown | null | undefined,
-    ) => void = (newValue, cb) => {
-        this.props.onChange({currentValue: newValue}, cb);
-        this.props.trackInteraction();
-    };
-
-    _handleFocus: () => void = () => {
-        this.props.onFocus([]);
-        this.setState((currentState) => {
-            return {...currentState, isFocused: true};
-        });
-    };
-
-    _handleBlur: () => void = () => {
-        this.props.onBlur([]);
-        this.setState((currentState) => {
-            return {...currentState, isFocused: false};
-        });
-    };
+            blur: () => {
+                if (inputRef.current) {
+                    inputRef.current.blur();
+                    setIsFocused(false);
+                }
+            },
+        }));
+
+        const handleChange = (
+            newValue: string,
+            cb?: () => unknown | null | undefined,
+        ): void => {
+            props.onChange({currentValue: newValue}, cb);
+            props.trackInteraction();
+        };
 
-    render(): React.ReactNode {
-        let labelText = this.props.labelText;
-        if (labelText == null || labelText === "") {
-            labelText = this.context.strings.yourAnswerLabel;
-        }
+        const handleFocus = (): void => {
+            props.onFocus([]);
+            setIsFocused(true);
+        };
 
-        // To right align a custom keypad we need to wrap it.
-        const maybeRightAlignKeypadInput = (
-            keypadInput: React.ReactElement<
```

Comment:
I felt that this was needlessly complicating the code readability, so I've opted to simply add the div around both left/right aligned inputs that gains a class if it needs to be right aligned.